plugin/inlay_hint.py

```diff
@@ -0,0 +1,153 @@
+        # Insert textEdits for the given inlay hint.
+        # If a InlayHintLabelPart was clicked, label_part will be passed as an argument to the LspInlayHintClickCommand
+        # and InlayHintLabelPart.command will be executed.
+        session = self.session_by_name(session_name, 'inlayHintProvider')
+        if session and session.has_capability('inlayHintProvider.resolveProvider'):
+            request = Request.resolveInlayHint(inlay_hint, self.view)
+            session.send_request_async(
+                request,
+                lambda response: self.handle(session_name, response, phantom_uuid, label_part))
+            return
+        self.handle(session_name, inlay_hint, phantom_uuid, label_part)
+
+    def handle(self, session_name: str, inlay_hint: InlayHint, phantom_uuid: str,
+               label_part: Optional[InlayHintLabelPart] = None) -> None:
+        self.handle_inlay_hint_text_edits(session_name, inlay_hint, phantom_uuid)
+        self.handle_label_part_command(session_name, label_part)
+
+    def handle_inlay_hint_text_edits(self, session_name: str, inlay_hint: InlayHint, phantom_uuid: str) -> None:
+        session = self.session_by_name(session_name, 'inlayHintProvider')
+        if not session:
+            return
+        text_edits = inlay_hint.get('textEdits')
+        if not text_edits:
+            return
+        for sb in session.session_buffers_async():
+            sb.remove_inlay_hint_phantom(phantom_uuid)
+        apply_text_edits_to_view(text_edits, self.view)
+
+    def handle_label_part_command(self, session_name: str, label_part: Optional[InlayHintLabelPart] = None) -> None:
+        if not label_part:
+            return
+        command = label_part.get('command')
+        if not command:
+            return
+        args = {
+            "session_name": session_name,
+            "command_name": command["command"],
+            "command_args": command["arguments"]
+        }
+        self.view.run_command("lsp_execute", args)
+
+
+def inlay_hint_to_phantom(view: sublime.View, inlay_hint: InlayHint, session: Session) -> sublime.Phantom:
+    position = inlay_hint["position"]  # type: ignore
+    region = sublime.Region(point_to_offset(Point.from_lsp(position), view))
+    phantom_uuid = str(uuid.uuid4())
+    content = get_inlay_hint_html(view, inlay_hint, session, phantom_uuid)
+    p = sublime.Phantom(region, content, sublime.LAYOUT_INLINE)
+    setattr(p, 'lsp_uuid', phantom_uuid)
+    return p
+
+
+def get_inlay_hint_html(view: sublime.View, inlay_hint: InlayHint, session: Session, phantom_uuid: str) -> str:
+    tooltip = format_inlay_hint_tooltip(inlay_hint.get("tooltip"))
+    label = format_inlay_hint_label(inlay_hint, session, phantom_uuid)
+    margin_left = "0.6rem" if inlay_hint.get("paddingLeft", False) else "0"
+    margin_right = "0.6rem" if inlay_hint.get("paddingRight", False) else "0"
```

Comment:
I have now tested this a bit more and I don't think this looks good. The phantoms are spaced too much from the code:
![Screenshot 2022-08-19 at 23 34 22](https://user-images.githubusercontent.com/153197/185710298-cfcd2d56-9923-452c-bbe2-38987b0fd8f7.png)
 
The likely reason for this is that ST by itself adds a small margin already so I don't think it makes sense to apply more, even if paddingLeft or paddingRight is set.

Reply:
VSCode for comparision. Se how the first hint (`title`) is just next to the parenthesis and has reasonable (not too big) margin on the right.

![Screenshot 2022-08-19 at 23 37 30](https://user-images.githubusercontent.com/153197/185710669-260060de-3181-4b01-a97f-7031273dba0c.png)